test/integration/webhook/dynamic_source_test.go

```diff
@@ -45,10 +45,10 @@ import (
 // Ensure that when the source is running against an apiserver, it bootstraps
 // a CA and signs a valid certificate.
 func TestDynamicSource_Bootstrap(t *testing.T) {
-	ctx, cancel := context.WithTimeout(logr.NewContext(context.Background(), testr.New(t)), time.Second*40)
+	ctx, cancel := context.WithTimeout(logr.NewContext(t.Context(), testr.New(t)), time.Second*40)
```

Comment:
I think we can remove a lot of these cancellable contexts, as `t.Context()` is cancelled automatically.